File: 0206-reverse-linked-list/0206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {

        ListNode tail = null;
        ListNode current = head;

        while(current != null){
            ListNode temp = current.next;
            current.next = tail;
            tail = current;
            current = temp;
        }

        return tail;
    }
}

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:

        #Time : O(N)
        #Space : O(1)
        
        tail = None

        current = head
        while current:
            temp = current.next
            current.next = tail 
            tail = current
            current = temp
        
        return tail